### ISSUE TITLE
Don't ignore PGP for packages installed from command line

### ISSUE THREAD

**Issue by voxik** [rpm-software-management/dnf5#991]
~~~
$ sudo dnf update https://kojipkgs.fedoraproject.org//packages/gnome-control-center/45.1/2.fc40/data/signed/a15b79cc/x86_64/gnome-control-center-45.1-2.fc40.x86_64.rpm https://kojipkgs.fedoraproject.org//packages/gnome-control-center/45.1/2.fc40/data/signed/a15b79cc/noarch/gnome-control-center-filesystem-45.1-2.fc40.noarch.rpm

... snip ...

Warning: skipped PGP checks for 2 package(s).
~~~

I don't see reason why PGP check should be ignored. I don't think DNF4 would ignore them.

~~~
$ rpm -q dnf5
dnf5-5.1.6-1.fc40.x86_64
~~~

**Comment by j-mracek** [comment]
I tested the reproducer with DNF4 and I can confirm that DNF5 behaves by the same way as DNF4. The only difference is that dnf4 does not report that there are packages installed without PGP verification.

If we will change the behavior I would prefer to do it with a FESCO approval. This change might break many workflows.

**Comment by ppisar** [comment]
I agree with voxik that it's an unfortunate behavior, especially if the package is not local. At the same time I agree with j-mracek that the requested change brings a significant change and that we should get an approval from a larger user base, e.g. from FESCO.

I can see a that users installing their local packages would have to explicitly pass --no-gpgchecks option and that's probably the reason why the check is implicitly disabled. A fun fact: There is no --gpgchecks option to opt for the check.

**Comment by ppisar** [comment]
I asked on Fedora devel mailing list <https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/thread/IS7WERNT6AGISWA5BZD2X2FFMSJHNZQJ/>.

**Comment by ppisar** [comment]
Kevin Fenzi came with DNF4 localpkg_gpgcheck option which controls checking signatures of local packages and which defaults to false. I confirm that this option exists and behaves alike in DNF5 too:

~~~~
# dnf5 --setopt=localpkg_gpgcheck=true install …
~~~~

As a workaround you can place "localpkg_gpgcheck=true" into /etc/dnf/dnf.conf and DNF5 will start requiring valid signatures for non-repository packages.

**Comment by ppisar** [comment]
I will keep a few weeks people to react on the mailing list and if there will be expressed a remarkable demand for a change of the behavior, we will propose the change to FESCo. Otherwise, we will close this request without any action.

**Comment by j-mracek** [comment]
There is a ticket that overlaps with this issue - https://github.com/rpm-software-management/dnf5/issues/727. I would suggest to resolve both issue at the same time.

**Comment by ppisar** [comment]
#727 is interesting. 
By the way localpkg_gpgcheck cannot be overridden with --no-gpgchecks. localpkg_gpgcheck has a stronger priority.

**Comment by mcrha** [comment]
Using dnf5-5.2.5.0-20240828005842.55.g4b2fb499.fc40.x86_64.

I've upgraded the machine from f39 to f40.

When I run `dnf5 update` in the f40 it asks to import the f40 PGP key, as expected.

When I run `dnf5 update --offline`, it does _not_ ask to import the key. It claims on the terminal:

> Warning: skipped PGP checks for 2 packages from repository: @stored_transaction

Similarly when running an offline update using the D-Bus API of the dnf5daemon-server; it does not ask to import the key too.

I think it makes sense to have the PGP key imported and checked for (when the repo has it enabled) regardless whether the update is done immediately or offline.

_If you think this bug is different from that mine, then just say so and I can create a new bug instead._

**Comment by jan-kolarik** [comment]
> Using dnf5-5.2.5.0-20240828005842.55.g4b2fb499.fc40.x86_64.
> 
> I've upgraded the machine from f39 to f40.
> 
> When I run `dnf5 update` in the f40 it asks to import the f40 PGP key, as expected.
> 
> When I run `dnf5 update --offline`, it does _not_ ask to import the key. It claims on the terminal:
> 
> > Warning: skipped PGP checks for 2 packages from repository: @stored_transaction
> 
> Similarly when running an offline update using the D-Bus API of the dnf5daemon-server; it does not ask to import the key too.
> 
> I think it makes sense to have the PGP key imported and checked for (when the repo has it enabled) regardless whether the update is done immediately or offline.
> 
> _If you think this bug is different from that mine, then just say so and I can create a new bug instead._

Thanks for the report. I am not sure right now if it's really the same cause, so I've moved this into a separate issue https://github.com/rpm-software-management/dnf5/issues/1668.
